src/clients/stormGlass.ts

```diff
@@ -75,10 +77,30 @@ export class StormGlass {
     'swellDirection,swellHeight,swellPeriod,waveDirection,waveHeight,windDirection,windSpeed';
   readonly stormGlassAPISource = 'noaa';
 
-  constructor(protected request = new HTTPUtil.Request()) {}
+  constructor(
+    protected request = new HTTPUtil.Request(),
+    protected cacheUtil = new CacheUtil()
```

Comment:
I was thinking about this, here we are starting a new CacheUtil every time that the StormGlass client is instantiated, it's not wrong and I know node-cache is simple but imagine we change it to redis for example, we would keep the `CacheUtil` probably, and just replace the NodeCache to Redis.
The problem is that redis, Memcached and etc, they open a connection, if we keep initializing a new class many times it will result in having multiple connections opened and can lead to memory leaks or even reaching the limit of connections.

For this sort of services I prefer the same approach we do for databases, just instantiate it once and use the same object everywhere. The change here is basically to change the `CacheUtil` to be a singleton:

1. Change the `src/util/cache.ts`to return by default a new instance of the CacheUtil, like `export default new CacheUtil()` it will force Nodejs to instantiate it just once, the first time the module is imported it will get instantiated and any subsequent import will use the same instance from the node modules cache.
2. Change the `import { CacheUtil } from '@src/util/cache';` to `import CacheUtil from '@src/util/cache';`
3. Change `protected cacheUtil = new CacheUtil()` to `protected cacheUtil = CacheUtil`

And done! now we use just one instance! 
I have a video about this pattern here https://www.youtube.com/watch?v=MPbybQdqcrE

If you have any question feel free to ask! Again it's not wrong, but I think we should leave to code able to handle different cache providers in the future in case we decide to add a redis for example.

Reply:
That's make all sense! I have already watched this video but I didn't think to use it in this case. Thanks for suggest it!